### DISCUSSION TITLE
Setting RLS when not using Supabase for authentication

### DISCUSSION THREAD

**Discussion by mohnish7** [supabase/supabase#3300] [Questions]
Hello! I'm hoping to create new policies for inserting/updating data so that I can continue using RLS on my tables. I am not using Supabase for authentication, and instead would only like to limit insert/update access to people with the same company_id (which is stored in the object returned from the authentication service I'm using). How would this be possible?

My query looks like this:
`await supabase.from('table_name').select('column').eq('company_id', obj.company_id)`

**Comment by dshukertjr** [comment]
Hi @mohnish7!

If you are not using Supabase Auth, I don't think there is a way of limiting access using RLS as of now. 

Would moving your authentication to Supabase be possible? I think if that is possible, it would be the easiest and cleanest way of doing this. 

If using Supabase auth is not possible, then you can maybe do the following:
1. Hide your data behind RLS so that no one can access it
2. Create your own back end with a end point to retrieve company info (or whatever data you want people to retrieve)
3. In that custom end point, whenever user calls it, verify the user using your custom auth and once verified, use [service key](https://supabase.io/docs/guides/api#details-and-links) to bypass RLS to get the necessary data and return it to the user
